### PR TITLE
feat: waku2 - use identify protocol

### PR DESCRIPTION
**Pull request by richard-ramos**
**BREAKING CHANGE**

This would use identify protocol to determine what protocols are supported by the waku2 peers, instead of having separate buckets in the configuration.

The node config in desktop and mobile should be changed from:

```js
ClusterConfig: {
    RelayNodes: ["/ip4/127.0.0.1/udp/1234",  "enstree:somednsdiscoveryurl..."],
    StoreNodes: ["/ip4/127.0.0.1/udp/1234",  "enstree:somednsdiscoveryurl..."],
    FilterNodes: ["/ip4/127.0.0.1/udp/1234",  "enstree:somednsdiscoveryurl..."],
    LightpushNodes: ["/ip4/127.0.0.1/udp/1234",  "enstree:somednsdiscoveryurl..."],
    WakuRendezvousNodes: ["/ip4/127.0.0.1/udp/1234",  "enstree:somednsdiscoveryurl..."]
}
```

to

```js
ClusterConfig: {
    WakuNodes: ["/ip4/127.0.0.1/udp/1234",  "enstree:somednsdiscoveryurl..."]
}
```

cc: @jm-clius @oskarth

### CODE REVIEW

File: wakuv2/waku.go [hunk 1]
```diff
@@ -317,51 +303,96 @@ func (w *Waku) dnsDiscover(enrtreeAddress string, protocol libp2pproto.ID, apply
...
+
+	err = identifyService.Host.Connect(ctx, *peerInfo)
+	if err != nil {
+		log.Error("could not extract peerinfo", zap.Any("ma", ma), zap.Error(err))
+		return
+	}
+
+	conns := identifyService.Host.Network().ConnsToPeer(peerInfo.ID)
+	if len(conns) == 0 {
+		return // No connection
```
Comment (alaibe) [inline wakuv2/waku.go:424]: shall we return an error in that case?

**Review by alaibe — APPROVED** [review]

File: wakuv2/waku.go (hunk 1 above)
Comment (richard-ramos) [reply wakuv2/waku.go:424]: hm, maybe logging an error but returning an error i think it's not necessary. I'm not sure how would I handle a case where the connection to a peer was succesful, yet libp2p Network() says that there's no connection to that peer

**Comment by churik** [comment]
@richard-ramos still relevant?

**Comment by richard-ramos** [comment]
@churik yes. I'll rebase it today

**Pull request merged**